### PR TITLE
Add AVAX C Chain Abandon Seed

### PR DESCRIPTION
**Pull request by lvndry**
### 📝 Description

- Add abandon seed for AVAX C Chain
- Add `CryptoCurrencyId` in `types-cryptoassets`

### ❓ Context

- **Impacted projects**: `LLC`
- **Linked resource(s)**: `https://ledgerhq.atlassian.net/browse/LIVE-4261`

### ✅ Checklist

- [x] **Test coverage** 
- [x] **Atomic delivery** 
- [x] **No breaking changes** 

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

### CODE REVIEW

File: apps/ledger-live-desktop/release-notes.json [hunk 1]
```diff
@@ -1,4 +1,12 @@
 [
```
Comment (henrily-ledger) [inline apps/ledger-live-desktop/release-notes.json:1]: I don't think you need to push the release note

File: libs/ledger-live-common/src/families/hedera/tests/js-estimateMaxSpendable.test.ts [hunk 2]
```diff
@@ -20,7 +21,7 @@ const account: Account = {
   blockHeight: 0,
   currency: {
     type: "CryptoCurrency",
-    id: "",
+    id: "" as CryptoCurrencyId,
```
Comment (henrily-ledger) [inline libs/ledger-live-common/src/families/hedera/tests/js-estimateMaxSpendable.test.ts:24]: Shouldn't we put hedera if it's just an id we want, like this we do not need to cast ?

File: libs/ledgerjs/packages/cryptoassets/src/currencies.ts [hunk 3]
```diff
@@ -2493,7 +2497,8 @@ export const cryptocurrenciesById: Record<CryptoCurrencyId, CryptoCurrency> = {
     ],
     keywords: ["xlm", "stellar"],
   },
-  osmosis: {
+  // osmosi
+  osmo: {
```
Comment (henrily-ledger) [inline libs/ledgerjs/packages/cryptoassets/src/currencies.ts:2501]: Are we sure that we want to rename osmosis here ? Are we sure we prevent all side effect on renaming this ?

File: libs/ledgerjs/packages/cryptoassets/src/abandonseed.ts [hunk 4]
```diff
@@ -1,13 +1,16 @@
...
+import { CryptoCurrencyId } from "@ledgerhq/types-cryptoassets";
 
 /**
  * these are either "dead"/"burn" addresses OR "abandon" seed addresses.
  * These addresses are PUBLIC addresses
  * We use them for tests and also for dry-run estimations
  * DO NOT USE AS RECIPIENT OR SIGN TRANSACTIONS INTO THEM
  */
-const abandonSeedAddresses: Record<string, string> = {
+const abandonSeedAddresses: Partial<Record<CryptoCurrencyId, string>> = {
```
Comment (henrily-ledger) [inline libs/ledgerjs/packages/cryptoassets/src/abandonseed.ts:10]: Does this work, i am not sure ? 
{[k: CryptoCurrencyId] : string }
I feel like this is more readable than Partial<Record<CryptoCurrencyId, string>>

File: libs/ledgerjs/packages/cryptoassets/src/abandonseed.ts [hunk 5]
```diff
@@ -72,10 +75,12 @@ const abandonSeedAddresses: Record<string, string> = {
...
  */
+// TODO: signature should be  getAbandonSeedAddress(currencyId: CryptoCurrencyId)
 export const getAbandonSeedAddress = (currencyId: string): string => {
   invariant(
     abandonSeedAddresses[currencyId] !== undefined,
     `No abandonseed available for ${currencyId}`
   );
-  return abandonSeedAddresses[currencyId];
+
+  return abandonSeedAddresses[currencyId]!;
```
Comment (henrily-ledger) [inline libs/ledgerjs/packages/cryptoassets/src/abandonseed.ts:85]: I don't know this syntax, what it does ? 
the ! at the end ?

File: libs/ledgerjs/packages/types-cryptoassets/src/index.ts [hunk 6]
```diff
@@ -76,7 +203,7 @@ export type ExplorerView = {
 export type CryptoCurrency = CurrencyCommon & {
   type: "CryptoCurrency";
   // unique internal id of a crypto currency
-  id: string;
+  id: CryptoCurrencyId | "LBRY" | "groestcoin";
```
Comment (henrily-ledger) [inline libs/ledgerjs/packages/types-cryptoassets/src/index.ts:206]: why is LRBY and groestcoin not included in the list before ?

**Review by henrily-ledger — CHANGES_REQUESTED** [review]

File: libs/ledgerjs/packages/cryptoassets/src/abandonseed.ts (hunk 4 above)
Comment (lvndry) [reply libs/ledgerjs/packages/cryptoassets/src/abandonseed.ts:10]: Yes the syntax is correct and works as intentend
`{ [k in CryptoCurrencyId]: string }` is also valid however it requires **all** cryptocurrencyIds where here we only want a part of it thus the partial.
`Partial<{ [k in CryptoCurrencyId]: string }>` works fine but I prefer the `Record` syntax here

File: libs/ledgerjs/packages/cryptoassets/src/abandonseed.ts (hunk 5 above)
Comment (lvndry) [reply libs/ledgerjs/packages/cryptoassets/src/abandonseed.ts:85]: The `!` says to typescript "don't worry this value is never null or undefined".
I can attest it because of the `invariant` call before. Ideally we should do as I precised in the todo and change the signature from string to CryptoCurrencyId. I started this change and it impacted way too many files for this PR so I left it as a TODO

File: libs/ledgerjs/packages/cryptoassets/src/currencies.ts (hunk 3 above)
Comment (lvndry) [reply libs/ledgerjs/packages/cryptoassets/src/currencies.ts]: AFAIK no risk as the keys are only use to access the objects themselves from the ID

File: libs/ledgerjs/packages/types-cryptoassets/src/index.ts (hunk 6 above)
Comment (lvndry) [reply libs/ledgerjs/packages/types-cryptoassets/src/index.ts]: These are actually typos made in the past. In the object `cryptocurrenciesById` we want the key to be the same as the id.
eg:

```ts
{
   my_coin: {
      id: "my_coin"
   }
}
```

However for these coin the id and the key is different eg key is `groestlcoin` and id is `groestcoin`. To make it compliant with the typing I had to add these values as valid ids even though they are not valid keys

File: libs/ledgerjs/packages/cryptoassets/src/currencies.ts (hunk 3 above)
Comment (lvndry) [reply libs/ledgerjs/packages/cryptoassets/src/currencies.ts]: Could solve the issue differently so I reverted this part

File: libs/ledgerjs/packages/cryptoassets/src/currencies.ts [hunk 7]
```diff
@@ -3535,13 +3541,8 @@ export function findCryptoCurrencyByTicker(
...
-/**
- *
- * @param {*} id
- */
-export function findCryptoCurrencyById(
-  id: string
-): CryptoCurrency | null | undefined {
+// TODO: signature should be findCryptoCurrencyById(id: CryptoCurrencyId)
+export function findCryptoCurrencyById(id: string): CryptoCurrency {
   return cryptocurrenciesById[id];
```
Comment (gre) [inline libs/ledgerjs/packages/cryptoassets/src/currencies.ts:3546]: I'm not sure about this change. There is a reason this is a "find" and not a "get". Use the get version instead in case you are sure about the id

Comment (lvndry) [reply libs/ledgerjs/packages/cryptoassets/src/currencies.ts:3546]: Good point.
However don't you think it can be confusing to have both available ? What is the use case for looking for a currencyId that we don't support ?

File: apps/ledger-live-desktop/release-notes.json (hunk 1 above)
Comment (lvndry) [reply apps/ledger-live-desktop/release-notes.json:1]: Tbh don't know why it's been updated

Comment (lambertkevin) [reply apps/ledger-live-desktop/release-notes.json:1]: It's automatically added after each build of the app

File: libs/ledger-live-common/src/families/evm/__snapshots__/bridge.integration.test.ts.snap [hunk 8]
```diff
@@ -1,3263 +0,0 @@
-// Jest Snapshot v1, https://goo.gl/fbAQLP
```
Comment (lambertkevin) [inline libs/ledger-live-common/src/families/evm/__snapshots__/bridge.integration.test.ts.snap:1]: Why is this whole file removed ?

File: libs/ledger-live-common/src/families/evm/__tests__/rpc.unit.test.ts [hunk 9]
```diff
@@ -1,9 +1,9 @@
 import { AssertionError } from "assert";
-import { CryptoCurrency } from "@ledgerhq/types-cryptoassets";
+import { CryptoCurrency, CryptoCurrencyId } from "@ledgerhq/types-cryptoassets";
 import { DEFAULT_RETRIES_RPC_METHODS, withApi } from "../api/rpc.common";
 
 const fakeCurrency: Partial<CryptoCurrency> = {
-  id: "my_new_chain",
+  id: "my_new_chain" as CryptoCurrencyId,
```
Comment (lambertkevin) [inline libs/ledger-live-common/src/families/evm/__tests__/rpc.unit.test.ts:6]: Doesn't really make sense here, it's mocked data, and my_new_chain isn't contained in the `@ledgerhq/types-cryptoassets` lib

File: libs/ledger-live-common/src/families/hedera/tests/js-transaction.test.ts [hunk 10]
```diff
@@ -25,7 +26,7 @@ const account: Account = {
   blockHeight: 0,
   currency: {
     type: "CryptoCurrency",
-    id: "",
+    id: "" as CryptoCurrencyId,
```
Comment (lambertkevin) [inline libs/ledger-live-common/src/families/hedera/tests/js-transaction.test.ts:29]: Same here, doesn't seem normal to cast an empty string to a very specific list of currency ids

File: libs/ledger-live-common/src/mock/fixtures/cryptoCurrencies.ts [hunk 11]
```diff
@@ -1,12 +1,15 @@
...
+  CryptoCurrencyId,
+} from "@ledgerhq/types-cryptoassets";
 import type { Account } from "@ledgerhq/types-live";
 import { cryptocurrenciesById } from "@ledgerhq/cryptoassets";
 
 export function createFixtureCryptoCurrency(family: string): CryptoCurrency {
   return {
     type: "CryptoCurrency",
-    id: "testCoinId",
+    id: "testCoinId" as CryptoCurrencyId,
```
Comment (lambertkevin) [inline libs/ledger-live-common/src/mock/fixtures/cryptoCurrencies.ts:12]: Doesn't make sense here either

File: libs/ledgerjs/packages/cryptoassets/README.md [hunk 12]
```diff
@@ -130,14 +126,6 @@ Returns **(CryptoCurrency | null | [undefined](https://developer.mozilla.org/doc
...
 Returns **(CryptoCurrency | null | [undefined](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/undefined))** 
 
-### findCryptoCurrencyById
-
-#### Parameters
-
-*   `id` **any** 
-
-Returns **(CryptoCurrency | null | [undefined](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/undefined))** 
-
```
Comment (lambertkevin) [inline libs/ledgerjs/packages/cryptoassets/README.md:140]: Doesn't seem normal that those functions are being removed from the doc, especially in this PR.

File: libs/ledgerjs/packages/cryptoassets/README.md [hunk 13]
```diff
@@ -155,14 +143,6 @@ Returns **(CryptoCurrency | null | [undefined](https://developer.mozilla.org/doc
...
 Returns **[boolean](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Boolean)** 
 
-### getCryptoCurrencyById
-
-#### Parameters
-
-*   `id` **any** 
-
-Returns **CryptoCurrency** 
-
```
Comment (lambertkevin) [inline libs/ledgerjs/packages/cryptoassets/README.md:165]: Doesn't seem normal that those functions are being removed from the doc, especially in this PR.

File: libs/ledgerjs/packages/cryptoassets/src/currencies.ts [hunk 14]
```diff
@@ -3440,6 +3450,8 @@ export const cryptocurrenciesById: Record<CryptoCurrencyId, CryptoCurrency> = {
   },
 };
 
+console.log(cryptocurrenciesById);
```
Comment (lambertkevin) [inline libs/ledgerjs/packages/cryptoassets/src/currencies.ts:3453]: Console.log

File: libs/ledgerjs/packages/cryptoassets/src/currencies.ts (hunk 7 above)
Comment (lambertkevin) [reply libs/ledgerjs/packages/cryptoassets/src/currencies.ts:3552]: As gre said, it's not the same. 
`find` is a search, you hope to get something but could result with nothing.
`get` *should*  result with something.

Those are heavily used pieces of software in the live, please don't change it without a good reason. You will end up breaking everything this way.

**Review by lambertkevin — CHANGES_REQUESTED** [review]
As a general comment, your PR is 99% out of scope of what it should be doing... I know it's tempting to update things when you see them, I'm myself guilty of doing such, but keep it as minimal as possible.

Also, it happens very often now that you take the initiative to update/remove things that are very important and shouldn't be changed without proper advice. Please, don't modify things just because you don't like them.

File: libs/ledgerjs/packages/cryptoassets/src/currencies.ts (hunk 7 above)
Comment (lvndry) [reply libs/ledgerjs/packages/cryptoassets/src/currencies.ts:3552]: As of now I didn't change the function and I removed the TODO but I'm still wondering what's the use case for it in the app

File: libs/ledger-live-common/src/families/evm/__snapshots__/bridge.integration.test.ts.snap (hunk 8 above)
Comment (lvndry) [reply libs/ledger-live-common/src/families/evm/__snapshots__/bridge.integration.test.ts.snap:1]: I ran `pnpm common jest --update-snapshot` and this was removed

File: libs/ledgerjs/packages/cryptoassets/README.md (hunk 12 above)
Comment (lvndry) [reply libs/ledgerjs/packages/cryptoassets/README.md:140]: I ran `pnpm doc:ljs` it's not a manual update

Comment (lambertkevin) [reply libs/ledgerjs/packages/cryptoassets/README.md]: Yes because you removed that function, and you shouldn't.

File: libs/ledgerjs/packages/cryptoassets/README.md (hunk 13 above)
Comment (lambertkevin) [reply libs/ledgerjs/packages/cryptoassets/README.md]: https://github.com/LedgerHQ/ledger-live/pull/1713#discussion_r1015257179

File: libs/ledgerjs/packages/cryptoassets/README.md (hunk 12 above)
Comment (lambertkevin) [reply libs/ledgerjs/packages/cryptoassets/README.md]: I know but it still shouldn't be happening so there is an issue somewhere.

File: libs/ledger-live-common/src/families/evm/__snapshots__/bridge.integration.test.ts.snap (hunk 8 above)
Comment (lambertkevin) [reply libs/ledger-live-common/src/families/evm/__snapshots__/bridge.integration.test.ts.snap:1]: Well that's part of my more general comment, I don't get why you update a snapshot in a PR supposed to just add a abandon seed, it shouldn't be related.

Comment (lvndry) [reply libs/ledger-live-common/src/families/evm/__snapshots__/bridge.integration.test.ts.snap:1]: Actually it's related to this action that had an incorrect snapshot: https://github.com/LedgerHQ/ledger-live/actions/runs/3360932223/jobs/5570716730#step:9:5122

File: apps/ledger-live-desktop/release-notes.json (hunk 1 above)
Comment (lvndry) [reply apps/ledger-live-desktop/release-notes.json:1]: So we should revert this change everytime we build the app locally ?

Comment (haammar-ledger) [reply apps/ledger-live-desktop/release-notes.json:1]: @lvndry correct, you can actually ignore this file locally with:
`git update-index --skip-worktree apps/ledger-live-desktop/release-notes.json`
and if needed undo with:
`git update-index --no-skip-worktree apps/ledger-live-desktop/release-notes.json`

It can be reverted for this PR indeed.

File: libs/ledgerjs/packages/types-cryptoassets/src/index.ts [hunk 15]
```diff
@@ -76,7 +202,7 @@ export type ExplorerView = {
 export type CryptoCurrency = CurrencyCommon & {
   type: "CryptoCurrency";
   // unique internal id of a crypto currency
-  id: string;
+  id: CryptoCurrencyId | "LBRY" | "groestcoin" | "osmo";
```
Comment (haammar-ledger) [inline libs/ledgerjs/packages/types-cryptoassets/src/index.ts:205]: Doesn't seem clean to have to define exception values here. What's the reason for not having them in `CryptoCurrencyId`?

File: libs/ledgerjs/packages/cryptoassets/src/currencies.ts [hunk 16]
```diff
@@ -3604,6 +3610,7 @@ export const findCryptoCurrencyByManagerAppName = (
 export const hasCryptoCurrencyId = (id: string): boolean =>
   id in cryptocurrenciesById;
 
+// TODO: signature should be getCryptoCurrencyById(id: CryptoCurrencyId)
```
Comment (haammar-ledger) [inline libs/ledgerjs/packages/cryptoassets/src/currencies.ts:3613]: What's preventing this, at the moment?

File: libs/ledgerjs/packages/cryptoassets/src/types.ts [hunk 17]
```diff
@@ -1,5 +1,7 @@
+import { CryptoCurrencyId } from "@ledgerhq/types-cryptoassets";
+
 export type ERC20Token = [
```
Comment (haammar-ledger) [inline libs/ledgerjs/packages/cryptoassets/src/types.ts:3]: While we're here, any reason for not moving this type to `types-cryptoassets`?

File: libs/ledgerjs/packages/cryptoassets/src/abandonseed.ts [hunk 18]
```diff
@@ -72,10 +75,12 @@ const abandonSeedAddresses: Record<string, string> = {
  * DO NOT USE AS RECIPIENT OR SIGN TRANSACTIONS INTO THEM
  * @param {*} currencyId
  */
+// TODO: signature should be  getAbandonSeedAddress(currencyId: CryptoCurrencyId)
```
Comment (haammar-ledger) [inline libs/ledgerjs/packages/cryptoassets/src/abandonseed.ts:78]: Same question, what's preventing this currently?

File: libs/ledgerjs/packages/types-cryptoassets/src/index.ts [hunk 19]
```diff
@@ -1,5 +1,131 @@
 import { CoinType } from "./slip44";
 
+// All cryptocurrency ids
+// list should be append only. **do not modify existing ids**
+export type CryptoCurrencyId =
```
Comment (haammar-ledger) [inline libs/ledgerjs/packages/types-cryptoassets/src/index.ts:5]: @gre what do you think of this change?

Idea is that `CryptoCurrencyId` was still defined in the `cryptoassets` package, which needed to be imported even when we only needed the IDs. So we better define it in `types-cryptoassets`.

File: libs/ledger-live-common/src/families/evm/__tests__/rpc.unit.test.ts (hunk 9 above)
Comment (haammar-ledger) [inline libs/ledger-live-common/src/families/evm/__tests__/rpc.unit.test.ts:6]: Does this work? As `my_new_chain` isn't part of the defined IDs.

File: apps/ledger-live-desktop/src/live-common-set-supported-currencies.js [hunk 20]
```diff
@@ -5,6 +5,7 @@ import { setPlatformVersion } from "@ledgerhq/live-common/platform/version";
 setPlatformVersion("1.1.0");
 
 setSupportedCurrencies([
+  "avalanche_c_chain",
```
Comment (haammar-ledger) [inline apps/ledger-live-desktop/src/live-common-set-supported-currencies.js:8]: Doesn't hurt, but not sure this is in the scope of this PR.
If you enable it for LLD, you should enable it as well for LLM + CLI + tests.
Or keep this for another PR.

**Review by haammar-ledger — COMMENTED** [review]
@lvndry Many changes seem out of the scope of this PR, right?

Doesn't hurt, but maybe you want to review how you split the changes between your branches?

File: libs/ledgerjs/packages/cryptoassets/src/currencies.ts (hunk 16 above)
Comment (lvndry) [reply libs/ledgerjs/packages/cryptoassets/src/currencies.ts:3613]: It's impacting a lot of code way outside of the scope of this PR

File: libs/ledgerjs/packages/cryptoassets/src/types.ts (hunk 17 above)
Comment (lvndry) [reply libs/ledgerjs/packages/cryptoassets/src/types.ts:3]: No reason I just didn't want to add changes not related to AVAX inte

File: libs/ledgerjs/packages/cryptoassets/src/abandonseed.ts (hunk 18 above)
Comment (lvndry) [reply libs/ledgerjs/packages/cryptoassets/src/abandonseed.ts:78]: A lot of impact outside of AVAX, I started but then reverted at the sight of files changed

File: apps/ledger-live-desktop/src/live-common-set-supported-currencies.js (hunk 20 above)
Comment (lvndry) [reply apps/ledger-live-desktop/src/live-common-set-supported-currencies.js:8]: I enabled for LLM, CLI and test in an other PR but didn't add this one
These are already merged in https://github.com/LedgerHQ/ledger-live/tree/feat/live-3999-coin-integration-avax

File: libs/ledger-live-common/src/families/evm/__tests__/rpc.unit.test.ts (hunk 9 above)
Comment (lvndry) [reply libs/ledger-live-common/src/families/evm/__tests__/rpc.unit.test.ts:6]: It does work indeed, it's a way to make typescript compiler consider `my_new_chain` as a CrytoCurrencyId

**Comment by lvndry** [comment]
Most changes are related to typing improvement initiatives started with the AVAX integration, so when changing a type sometimes other files are impacted but still related to the PR

File: libs/ledgerjs/packages/types-cryptoassets/src/index.ts (hunk 15 above)
Comment (lvndry) [reply libs/ledgerjs/packages/types-cryptoassets/src/index.ts:205]: It's because `cryptocurrenciesById` is of type `Record<CryptoCurrencyId, CryptoCurrency>`, and for most cryptocurrencies the key is the same as the `id` expect for these 3 exceptions.
So in the object we have

```ts
{
  lbry: {
      id: "LBRY",
      ...
  },
  groestlcoin: {
      id: "groestcoin", // maybe a typo we didn't notice
  },
  osmosis: {
      id: "osmo",
      ...
  }
} 
```
As we don't want to change currency ids I added this union
For future integration we'd only have to add the id in `CryptoCurrencyId`. this will avoid typos like for `groestlcoin` for exemple
